Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -821,12 +802,33 @@ extension NavigationViewController: NavigationServiceDelegate {
         let componentsWantAdvance = navigationComponents.allSatisfy { $0.navigationService(service, didArriveAt: waypoint) }
         let advancesToNextLeg = componentsWantAdvance && (delegate?.navigationViewController(self, didArriveAt: waypoint) ?? defaultBehavior)
         
-        arrivalController?.showEndOfRouteIfNeeded(self,
-                                                  advancesToNextLeg: advancesToNextLeg,
-                                                  onDismiss: { [weak self] in
-                                                    self?.navigationService.endNavigation(feedback: $0)
-                                                    self?.handleCancelAction()
-                                                  })
+        if advancesToNextLeg &&
+            router.routeProgress.isFinalLeg &&
+            showsEndOfRouteFeedback {
+            let destination = route?.legs.last?.destination
+            let controller = showEndOfRouteFeedbackCollection(destination: destination,
+                                                              onDismiss: { [weak self] in
+                                                                self?.navigationService.endNavigation(feedback: $0)
+                                                                self?.handleCancelAction()
+                                                              })
+            
+            navigationMapView?.navigationCamera.stop()
```

Comment:
Had to spit out camera logic here, outside `EndOfRouteFeedbackCollecting`, to separate the functionality. Hopefully this will be refined later, while tweaking `CameraController`.